rskj-core/src/main/java/co/rsk/db/importer/provider/index/BootstrapIndexCandidateSelector.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.db.importer.provider.index;
+
+import co.rsk.db.importer.BootstrapImportException;
+import co.rsk.db.importer.provider.index.data.BootstrapDataEntry;
+import co.rsk.db.importer.provider.index.data.BootstrapDataIndex;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class BootstrapIndexCandidateSelector {
+
+    private final List<String> publicKeys;
+    private final int minimumRequiredSources;
+
+    public BootstrapIndexCandidateSelector(List<String> publicKeys, int minimumRequiredSources) {
+        this.publicKeys = new ArrayList<>(publicKeys);
+        this.minimumRequiredSources = minimumRequiredSources;
+    }
+
+    public HeightCandidate getHeightData(List<BootstrapDataIndex> indexes) {
+        Map<Long, Map<String, BootstrapDataEntry>> entriesPerHeight = getEntriesPerHeight(indexes);
+        return getHeightCandidate(entriesPerHeight);
+    }
+
+    private HeightCandidate getHeightCandidate(Map<Long, Map<String, BootstrapDataEntry>> entriesPerHeight) {
+        HeightCandidate candidate = null;
+        for (Map.Entry<Long, Map<String, BootstrapDataEntry>> entry : entriesPerHeight.entrySet()) {
+            Long height = entry.getKey();
+            boolean isPossibleCandidate = candidate == null || candidate.getHeight() < height;
+            if (!isPossibleCandidate) {
+                continue;
+            }
+
+            Map<String, BootstrapDataEntry> entries = entry.getValue();
+            Map<String, Long> hashGroups = entries.values().stream()
+                    .collect(Collectors.groupingBy(BootstrapDataEntry::getHash, Collectors.counting()));
+            Optional<Map.Entry<String, Long>> bestHash = hashGroups.entrySet().stream()
+                    .max(Comparator.comparing(Map.Entry::getValue));
+
+            if (bestHash.isPresent() && bestHash.get().getValue() >= minimumRequiredSources) {
```

Comment:
Which condition should trigger the `bestHash.get().getValue > minimunRequiredSources`? I think we should change this to an equals and present a warning and skip the result if it is bigger than... 

I think bigger than means:
1. exporter ran twice on the same block
2. the signer did not care and signed something which was already in the index

those two things should not happen. if they do, something is broken.